### ISSUE TITLE
cargo clean required between cargo kani invocations

### ISSUE THREAD

**Issue by zhassan-aws** [model-checking/kani#1591]
I often run into issues when rerunning `cargo kani` without doing `cargo clean` in between.

One of the issues is the `Error: Conflicting proof harnesses named ...` error. I could reproduce through updating the dependencies or through adding a lib crate to a bin crate.

For example:
1. `cargo new --lib foo`
2. `cd foo`
3. Replace the contents of `src/lib.rs` with:
```rust
#[kani::proof]
fn check() {
    assert!(1 + 1 == 2);
}
```
4. Run `cargo kani --harness check`
5. Add a dependency to `Cargo.toml`, e.g.:
```toml
[dependencies]
bitflags = "1.3.2"
```
6. Rerun `cargo kani --harness check`

This produces:
```
   Compiling bitflags v1.3.2
   Compiling foo v0.1.0 (/home/ubuntu/examples/conflicting/tmp/foo)
    Finished dev [unoptimized + debuginfo] target(s) in 0.17s
Error: Conflicting proof harnesses named check:
 check
 check
```
The same error can be reproduced through creating a binary crate, runing `cargo kani --harness check`, adding a `src/lib.rs` file, and rerunning `cargo kani --harness check`.

This error does not occur if I `cargo clean` in between the two `cargo kani` invocations.

Apparently, the modification (adding a dependency or adding a module) results in changing the checksum, so the metadata files do not get overwritten. With both metadata files lying around, the Kani driver thinks that there are duplicate harnesses.

**Comment by zhassan-aws** [comment]
If I invoke `cargo kani` without `--harness` but edit the harness body in between invocations, I get an error in CBMC instead (the same error in https://github.com/model-checking/kani/issues/1448).

**Comment by zhassan-aws** [comment]
The root cause of both issues is that the Kani driver collects the symtab and metadata files through a * glob.

**Comment by tedinski** [comment]
Related #1478

**Comment by jaisnan** [comment]
Also another side effect of this issue, the output from `cargo kani` repeats itself without `cargo clean` before invocations.

**Comment by celinval** [comment]
This has been fixed by #1847
